SpatialGDK/Source/SpatialGDK/Public/Interop/SpatialRPCService.h

```diff
@@ -44,21 +47,31 @@ struct EntityRPCType
 
 enum class EPushRPCResult : uint8
 {
-	Success,
-
-	QueueOverflowed,
-	DropOverflowed,
-	HasAckAuthority,
-	NoRingBufferAuthority
+	None = 0x00,
```

Comment:
The enum stores different blocks of flags similar to how EObjectFlags works in the Engine.

Reply:
One of the key parts of this change is separating the reason for a failure and the action to take because of that failure.

Reply:
I think I understand now, thanks for your replies @cmsmithio!
Now I'm wondering, why do we not have 2 enums: `EPushRPCResult` and `EPushRPCAction` for example? I'd expect it would be easier to follow? Is it because you'd prefer to have 1 return variable for PushRPC?

Reply:
I considered that approach too (and we could certainly modify this change to work that way).  The primary reason was to future proof against needing a 3rd...Nth enum if we decide to add more info to the result of an RPC push attempt in future.  I haven't used an enum like this as a bitmask before, but since it was in the engine I wanted to give it a try, and hear people's thoughts on the result.